core/isolate.rs

```diff
@@ -1012,6 +1006,60 @@ pub mod tests {
     })
   }
 
+  #[test]
+  fn dyn_import_err2() {
+    use std::convert::TryInto;
+    // Import multiple modules to demonstrate that after failed dynamic import
+    // another dynamic import can still be run
+    run_in_task(|| {
+      let count = Arc::new(AtomicUsize::new(0));
+      let count_ = count.clone();
+      let mut isolate = Isolate::new(StartupData::None, false);
+      isolate.set_dyn_import(move |_, specifier, referrer| {
+        let c = count_.fetch_add(1, Ordering::Relaxed);
+        match c {
+          0 => assert_eq!(specifier, "foo1.js"),
+          1 => assert_eq!(specifier, "foo2.js"),
+          2 => assert_eq!(specifier, "foo3.js"),
+          _ => unreachable!(),
+        }
+        assert_eq!(referrer, "dyn_import_error.js");
+
+        let source_code_info = SourceCodeInfo {
+          module_url_specified: specifier.to_owned(),
+          module_url_found: specifier.to_owned(),
+          code: "# not valid JS".to_owned(),
```

Comment:
@ry I updated the test and verified that it fails on master. This bit `# not valid JS` represents the same situation as when that panic was encountered, ie. importing non-JS file